### PR TITLE
Update Linux-CIP to kernel version 6.1

### PR DESCRIPTION
**Pull request by diogo-ivo**
This pull request is an initial port to version 6.1 of linux-cip. Basic functionality seems to be working, including ethernet with the upstream version of the prueth driver. Some of the patches from 5.10 still need to be addressed, like EFI and some more advanced ethernet functionality.

### CODE REVIEW

File: recipes-kernel/linux/linux-iot2050-6.1.inc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    files_dir = os.path.join(d.getVar('THISDIR'), 'files')
+    patch_glob = os.path.join(files_dir, patchdir, '*.patch')
+    patches = glob.glob(patch_glob)
+    patches.sort()
+
+    return ' '.join(['file://' + patch[len(files_dir)+1:] for patch in patches])
+
+SRC_URI += " \
+    https://git.kernel.org/pub/scm/linux/kernel/git/cip/linux-cip.git/snapshot/linux-cip-${PV}.tar.gz \
```
Comment (jan-kiszka) [inline recipes-kernel/linux/linux-iot2050-6.1.inc:21]: We have https://cdn.kernel.org/pub/linux/kernel/projects/cip/6.1/ now - requesting snapshots failed too often.

**Comment by jan-kiszka** [comment]
Ethernet is not working yet, driver does not bind to the device.

File: recipes-kernel/linux/files/patches-6.1/0036-arm64-dts-ti-iot2050-Add-icssg-prueth-nodes-for-PG1-.patch [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
++		       <&main_udmap 0xc107>, /* egress slice 1 */
++
++		       <&main_udmap 0x4100>, /* ingress slice 0 */
++		       <&main_udmap 0x4101>, /* ingress slice 1 */
++		       <&main_udmap 0x4102>, /* mgmnt rsp slice 0 */
++		       <&main_udmap 0x4103>; /* mgmnt rsp slice 1 */
++		dma-names = "tx0-0", "tx0-1", "tx0-2", "tx0-3",
++			    "tx1-0", "tx1-1", "tx1-2", "tx1-3",
++			    "rx0", "rx1",
++			    "rxmgm0", "rxmgm1";
```
Comment (jan-kiszka) [inline recipes-kernel/linux/files/patches-6.1/0036-arm64-dts-ti-iot2050-Add-icssg-prueth-nodes-for-PG1-.patch:87]: `"rxmgm0", "rxmgm1"` is gone in ti-linux-6.1.y, k3-am654-base-board.dts. Likely just a style difference.

**Comment by jan-kiszka** [comment]
My Ethernet issue has been understood and resolved - firmware bug: https://patchwork.ozlabs.org/project/uboot/patch/aa0e6927-e35c-48b7-b3fb-785059f48910@siemens.com/

File: recipes-kernel/linux/files/patches-6.1/0036-arm64-dts-ti-iot2050-Add-icssg-prueth-nodes-for-PG1-.patch (hunk 2 above)
Comment (jan-kiszka) [reply recipes-kernel/linux/files/patches-6.1/0036-arm64-dts-ti-iot2050-Add-icssg-prueth-nodes-for-PG1-.patch:87]: But looking that the DTS again, that delta in the ti-linux rather seems like a bug there. Let's keep things as-is.

**Comment by BaochengSu** [comment]
@diogo-ivo Any updates on this?

**Comment by jan-kiszka** [comment]
Just tested this in secure boot setup, and our reboot errors are back:
```
...
[  OK  ] Stopped systemd-remount-fs…ount Root and Kernel File Systems.
         Stopping tee-supplicant.service - TEE Supplicant...
[  OK  ] Stopped tee-supplicant.service - TEE Supplicant.
[  OK  ] Reached target shutdown.target - System Shutdown.
[  OK  ] Reached target final.target - Late Shutdown Services.
[  OK  ] Finished systemd-reboot.service - System Reboot.
[  OK  ] Reached target reboot.target - System Reboot.
[  693.436109] watchdog: watchdog0: nowayout prevents watchdog being stopped!
[  693.443675] watchdog: watchdog0: watchdog did not stop!
[  693.484356] systemd-shutdown[1]: Using hardware watchdog 'K3 RTI Watchdog', version 0, device /dev/watchdog0
[  693.494562] systemd-shutdown[1]: Modifying watchdog timeout is not supported, reusing the programmed timeout.
[  693.504586] systemd-shutdown[1]: Watchdog running with a timeout of 1min.
[  693.551009] systemd-shutdown[1]: Syncing filesystems and block devices.
[  693.579837] systemd-shutdown[1]: Sending SIGTERM to remaining processes...
[  693.600918] systemd-journald[1450]: Received SIGTERM from PID 1 (systemd-shutdow).
[  693.620817] systemd-shutdown[1]: Sending SIGKILL to remaining processes...
[  693.641910] systemd-shutdown[1]: Unmounting file systems.
[  693.650484] (sd-remount)[1829]: Remounting '/etc' read-only with options 'lowerdir=/root/etc,upperdir=/root/var/local/etc,workdir=/root/var/local/.etc-atomic'.
[  693.676796] (sd-umount)[1830]: Unmounting '/etc'.
[  693.705311] EXT4-fs (dm-2): unmounting filesystem.
[  693.742062] systemd-shutdown[1]: All filesystems unmounted.
[  693.747690] systemd-shutdown[1]: Deactivating swaps.
[  693.752764] systemd-shutdown[1]: All swaps deactivated.
[  693.758000] systemd-shutdown[1]: Detaching loop devices.
[  693.766053] systemd-shutdown[1]: All loop devices detached.
[  693.771654] systemd-shutdown[1]: Stopping MD devices.
[  693.777039] systemd-shutdown[1]: All MD devices stopped.
[  693.782362] systemd-shutdown[1]: Detaching DM devices.
[  693.789535] systemd-shutdown[1]: Detaching DM /dev/dm-2 (252:2).
[  693.834609] systemd-shutdown[1]: Detaching DM /dev/dm-1 (252:1).
[  693.886590] systemd-shutdown[1]: Not all DM devices detached, 1 left.
[  693.893379] systemd-shutdown[1]: Detaching DM devices.
[  693.899806] systemd-shutdown[1]: Not all DM devices detached, 1 left.
[  693.906502] systemd-shutdown[1]: Detaching DM devices.
[  693.912544] systemd-shutdown[1]: Not all DM devices detached, 1 left.
[  693.919001] systemd-shutdown[1]: Cannot finalize remaining DM devices, continuing.
[  693.926574] watchdog: watchdog0: nowayout prevents watchdog being stopped!
[  693.933451] watchdog: watchdog0: watchdog did not stop!
[  693.943170] systemd-shutdown[1]: Failed to finalize DM devices, ignoring.
[  693.951167] systemd-shutdown[1]: Syncing filesystems and block devices.
[  693.958145] systemd-shutdown[1]: Rebooting.
E/TC:? 0 get_rpc_alloc_res:645 RPC allocation failed. Non-secure world result: ret=0xffff000c ret_origin=0x2
E/TC:? 0 get_rpc_alloc_res:645 RPC allocation failed. Non-secure world result: ret=0xffff000c ret_origin=0x2
E/TC:? 0 
E/TC:? 0 TA panicked with code 0xffff000c
E/LD:  Status of TA bc50d971-d4c9-42c4-82cb-343fb7f37896
E/LD:   arch: aarch64
E/LD:  region  0: va 0x40004000 pa 0x9ee1a000 size 0x002000 flags rw-s (ldelf)
E/LD:  region  1: va 0x40006000 pa 0x9ee1c000 size 0x008000 flags r-xs (ldelf)
E/LD:  region  2: va 0x4000e000 pa 0x9ee24000 size 0x001000 flags rw-s (ldelf)
E/LD:  region  3: va 0x4000f000 pa 0x9ee25000 size 0x004000 flags rw-s (ldelf)
E/LD:  region  4: va 0x40013000 pa 0x9ee29000 size 0x001000 flags r--s
E/LD:  region  5: va 0x40014000 pa 0x9eeb1000 size 0x011000 flags rw-s (stack)
E/LD:  region  6: va 0x40025000 pa 0xf2b9c000 size 0x002000 flags rw-- (param)
E/LD:  region  7: va 0x40044000 pa 0x00001000 size 0x068000 flags r-xs [0]
E/LD:  region  8: va 0x400ac000 pa 0x00069000 size 0x01f000 flags rw-s [0]
E/LD:   [0] bc50d971-d4c9-42c4-82cb-343fb7f37896 @ 0x40044000
E/LD:  Call stack:
E/LD:   0x40082c18
E/LD:   0x40044b58
E/LD:   0x400451d8
E/LD:   0x40064a40
E/LD:   0x4007dc0c
E/LD:   0x40045a84
E/LD:   0x40088ec8
E/LD:   0x4007ddbc
[  694.079466] tpm tpm0: ftpm_tee_tpm_op_send: SUBMIT_COMMAND invoke error: 0xffff3024
[  694.087146] tpm tpm0: tpm_try_transmit: send(): error -53212
[  694.093553] sd 0:0:0:0: [sda] Synchronizing SCSI cache
[  694.118914] reboot: Restarting system
```
I don't get it yet, tee-supplicant is shut down and should have disconnected the fTPM driver, but there is still some command issued that fails in the absence of the supplicant daemon.

**Comment by jan-kiszka** [comment]
Shutdown issue resolved via optee-client patch in isar, see https://groups.google.com/g/isar-users/c/T5FuJo9A2hc.

**Comment by BaochengSu** [comment]
@diogo-ivo DCO check failed, see the details:

Commit sha: [53eb5db](https://github.com/siemens/meta-iot2050/pull/489/commits/53eb5db5663286fafba343bce06d1af47573ad24), Author: Diogo Ivo, Committer: Diogo Ivo; Expected "Diogo Ivo [diogo.ivo@tecnico.ulisboa.pt](mailto:diogo.ivo@tecnico.ulisboa.pt)", but got "Diogo Ivo [diogo.ivo@siemens.com](mailto:diogo.ivo@siemens.com)".

**Comment by BaochengSu** [comment]
> @diogo-ivo DCO check failed, see the details:
> 
> Commit sha: [53eb5db](https://github.com/siemens/meta-iot2050/pull/489/commits/53eb5db5663286fafba343bce06d1af47573ad24), Author: Diogo Ivo, Committer: Diogo Ivo; Expected "Diogo Ivo [diogo.ivo@tecnico.ulisboa.pt](mailto:diogo.ivo@tecnico.ulisboa.pt)", but got "Diogo Ivo [diogo.ivo@siemens.com](mailto:diogo.ivo@siemens.com)".

I guess you might have to update your email in your account settings?

**Pull request closed without merging**